index.js

```diff
@@ -51,27 +44,38 @@ function uglifyify(file, opts) {
     )
 
     debug = opts.sourceMap !== false && (debug || matched)
-    opts  = extend({}, {
+
+    var thisopts  = extend({}, {
```

Comment:
this is bikeshedding, no?

sorry, I have lost my sense of humour about this, too much head-banging and time lost due to lazy/dumb/unfriendly option handling (yeah, hi there Babel!)

regardless, I think you should run with your PR's in favour of this one, the end result is the same in terms of fixing the relevant problems/bugs and you seem to be more involved (with the project). 

so, by al means delete/close this PR once your's is merged.